lib/ansible/plugins/action/template.py

```diff
@@ -45,8 +44,8 @@ def get_checksum(self, dest, all_vars, try_directory=False, source=None, tmp=Non
                 dest = os.path.join(dest, base)
                 dest_stat = self._execute_remote_stat(dest, all_vars=all_vars, follow=False, tmp=tmp)
 
-        except AnsibleError:
-            return dict(failed=True, msg=to_native(get_exception()))
+        except AnsibleError as e:
```

Comment:
I think I'd remove most of syntax compatibility changes from this pr/commit, and do those in a different commit/pr.

(At least in cases where there are also non-syntax related changes that may eventually need to be back-ported/cherry-picked. Granted, in this case, the non-syntax changes are also not  2.4 compatible so it is unlikely to ever get backported to a works-with-py24 branch).

But in the past I've found it useful to have syntax related changes isolated.